pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -755,6 +780,82 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(ctx context.Context,
 	return reconcile.Result{}, nil
 }
 
+// validatePVCTopologyCompatibility checks if the existing PVC's topology annotation is compatible
+// with the volume's actual placement zone.
+func validatePVCTopologyCompatibility(ctx context.Context, pvc *v1.PersistentVolumeClaim,
+	volumeDatastoreURL string, topologyMgr commoncotypes.ControllerTopologyService,
+	vc *cnsvsphere.VirtualCenter) error {
+	log := logger.GetLogger(ctx)
+
+	// Check if PVC has topology annotation
+	topologyAnnotation, exists := pvc.Annotations[common.AnnVolumeAccessibleTopology]
+	if !exists || topologyAnnotation == "" {
+		// No topology annotation on PVC, skip validation
+		log.Debugf("PVC %s/%s has no topology annotation, skipping topology validation",
+			pvc.Namespace, pvc.Name)
+		return nil
+	}
+
+	// Parse PVC topology annotation
+	var pvcTopologySegments []map[string]string
+	err := json.Unmarshal([]byte(topologyAnnotation), &pvcTopologySegments)
+	if err != nil {
+		return fmt.Errorf("failed to parse topology annotation on PVC %s/%s: %v",
+			pvc.Namespace, pvc.Name, err)
+	}
+
+	// Get volume topology from datastore URL
+	volumeTopologySegments, err := topologyMgr.GetTopologyInfoFromNodes(ctx,
+		commoncotypes.WCPRetrieveTopologyInfoParams{
+			DatastoreURL:        volumeDatastoreURL,
+			StorageTopologyType: "zonal",
```

Comment:
if we are registering volume present on shared datastore or corss-zonal datastore, then we should not be using hardcoded StorageTopologyType here.